### PR TITLE
[risk=no]Fix NPE in userHasFreeTierCredits

### PR DESCRIPTION
**Pull request by jmthibault79**
Add comments and tests, rename methods

Description:

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java [hunk 1]
```diff
@@ -61,10 +62,6 @@ public FreeTierBillingService(
     this.workbenchConfigProvider = workbenchConfigProvider;
   }
 
-  public boolean userHasFreeTierCredits(DbUser user) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java]: Renamed to `userHasRemainingFreeTierCredits` and moved below

File: api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java [hunk 2]
```diff
@@ -245,11 +258,7 @@ public Double getUserCachedFreeTierUsage(DbUser user) {
    * @return the US dollar amount, represented as a double
    */
   public double getUserFreeTierDollarLimit(DbUser user) {
-    final Double override = user.getFreeTierCreditsLimitDollarsOverride();
-    if (override != null) {
-      return override;
-    }
-
-    return workbenchConfigProvider.get().billing.defaultFreeCreditsDollarLimit;
+    return Optional.ofNullable(user.getFreeTierCreditsLimitDollarsOverride())
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java]: No functional change here

File: api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java [hunk 3]
```diff
@@ -230,13 +227,29 @@ private void maybeAlertOnCostThresholds(
    * have created. This is NOT live BigQuery data: it is only as recent as the last
    * checkFreeTierBillingUsage cron job, recorded as last_update_time in the DB.
    *
+   * <p>Note: return value may be null, to enable direct assignment to the nullable Profile field
+   *
    * @param user the user as represented in our database
    * @return the total USD amount spent in workspaces created by this user, represented as a double
    */
-  public Double getUserCachedFreeTierUsage(DbUser user) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java]: renamed: "user" is redundant

File: api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java [hunk 4]
```diff
@@ -230,13 +227,29 @@ private void maybeAlertOnCostThresholds(
...
+  @Nullable
+  public Double getCachedFreeTierUsage(DbUser user) {
     return workspaceFreeTierUsageDao.totalCostByUser(user);
   }
 
+  /**
+   * Does this user have remaining free tier credits? Compare the user-specific free tier limit (may
+   * be the system default) to the amount they have used.
+   *
+   * @param user the user as represented in our database
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java]: nit: can it be like user: the user as represented in our database to make it more readable

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspacesControllerTest.java [hunk 5]
```diff
@@ -236,7 +236,7 @@
           .domainId("Measurement")
           .countValue(256L)
           .prevalence(0.4F)
-          .conceptSynonyms(new ArrayList<String>());
+          .conceptSynonyms(new ArrayList<>());
```
Comment (NehaBroad) [inline api/src/test/java/org/pmiops/workbench/workspaces/WorkspacesControllerTest.java]: Why did we remove <String>?

File: api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java (hunk 4 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java]: I like that idea, but a space is the javadoc standard for `@param`

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspacesControllerTest.java (hunk 5 above)
Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/workspaces/WorkspacesControllerTest.java]: I have an auto-linter configured which does small fixes like this when I touch a file.

**Pull request merged**